src/phoenix/utilities/client.py

```diff
@@ -0,0 +1,82 @@
+import warnings
+from typing import Any
+
+import httpx
+
+PHOENIX_SERVER_VERSION_HEADER = "x-phoenix-server-version"
+
+
+class VersionedClient(httpx.Client):
+    """
+    A httpx.Client wrapper that warns if there is a server/client version mismatch.
+    """
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        from phoenix import __version__ as phoenix_version
+
+        super().__init__(*args, **kwargs)
+        self._client_phoenix_version = phoenix_version
+        self._major, self._minor, self._patch = map(int, self._client_phoenix_version.split("."))
+        self._warned_on_minor_version_mismatch = False
+
+    def _check_version(self, response: httpx.Response) -> None:
+        server_version = response.headers.get(PHOENIX_SERVER_VERSION_HEADER)
+        if server_version is None:
+            return
```

Comment:
We can assume the server version is less than the first server version returning version headers (4.15 or whatever it turns out to be). Otherwise, we won't be able to offer a warning message on currently deployed servers.

Reply:
I don't think it's worth adding extra logic for that honestly, I can add a warning if we get "None" though

Reply:
changed this to return a generic error if we get no server version in the header